### PR TITLE
Swarm command propagation, omq rpc, msg expiry & deletion

### PR DESCRIPTION
**Pull request by jagerman**
(Builds on top of #432)

This refactors RPC handling to unify it between HTTPS, onion requests, and (new) OMQ RPC requests, and to significantly clean up, abstract, and document what different commands do.

Breakdown of changes:
- Make random message retrieval more efficient (the current approach scales poorly as the database grows)
- Refactor how timestamps and TTLs are handled; we now store a timestamp and an expiry, but no longer the TTL value: instead the TTL value is simply a temporary value that can be given when storing a message to implicitly define the expiry.
- Remove dead code/disused OMQ endpoints.
- Fix SS not shutting down properly if it gets signalled while still trying to get the initial keys from oxend.
- Change database storage to store bytes rather than base64-encoded data.  Base64 is a transport encoding that is needed for javascript, but was being wastefully stored in the database too.
- Change generated hash to be based on fundamental values rather than user-provided encodings.
- Don't require timestamp and ttl to be passed as strings anymore.
- Allow storing using timestamp+expiry (as an alternative to timestamp+ttl).
- Expose storage rpc endpoints through a new public oxenmq rpc category `storage.WHATEVER` (e.g. `storage.store`).  This allows clients that want to use zmq speak more efficiently to SS.
- Add bt-encoded input/output support to the OMQ storage endpoints; this is noticeably more network efficient because it requires neither base64 encoding, nor establishing and handshaking new connections for every request.  The OMQ storage endpoints take params either json or a bt-encoded dict, and reply in kind.
- Clean up internals by moving transport encoding of internal values closer to the transport layer.  (Aside from a cleaner design, this was also needed to get bt-encoded responses out cleanly).
- Add delete/expiry API interfaces to delete all, deleted selected, delete by timestamp, shorten all expiries, and shorten specific message expiries.

Still TODO:
- [x] wire up actual deletion and expiry operations (currently just stubs)
- [x] propagate, collect, and coalesce responses from fellow swarm members
- [x] make the `store` call recursive (storage already is, but there is no confirmation and it should be adapted to use the new store mechanism + confirmations added herein)
- [x] add database migration the database (there are some unused columns, a completely broken index, some unfixable design problems relying on `rowid`, and some wasteful storage encoding).
- [x] add timestamp to all storage rpc returns so that the client doesn't have to fetch it unless they've been down for a long time.
- [x] more testing

### CODE REVIEW

File: crypto/src/oxend_key.cpp [hunk 1]
```diff
@@ -57,4 +60,34 @@ x25519_pubkey x25519_seckey::pubkey() const {
     return pk;
 };
 
+template <typename T>
+static T parse_pubkey(std::string_view pubkey_in) {
```
Comment (majestrate) [inline crypto/src/oxend_key.cpp:64]: return `std::optional<T>` ?

File: httpserver/service_node.cpp [hunk 2]
```diff
@@ -833,108 +851,121 @@ void ServiceNode::oxend_ping() {
...
+                [this, testee, item, height=block_height_]
+                (cpr::Response r) {
+                    auto& pk = testee.pubkey_legacy;
+                    std::string status;
+                    std::string answer;
+                    if (r.error.code != cpr::ErrorCode::OK)
+                        OXEN_LOG(debug, "FAILED storage test of {}: {}", pk, r.error.message);
+                    else if (r.status_code != 200)
+                        OXEN_LOG(debug, "FAILED storage test of {}: received non-200 status {}",
+                                pk, r.status_code, r.status_line);
```
Comment (majestrate) [inline httpserver/service_node.cpp]: missing `{}` or extra formatting argument

File: crypto/src/oxend_key.cpp (hunk 1 above)
Comment (jagerman) [reply crypto/src/oxend_key.cpp:64]: Not necessary for the pubkey types: they have a built-in `operator bool()` that returns false if set to an all-0 value.

File: httpserver/client_rpc_endpoints.h [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+/// - `pubkey` (required) contains the pubkey of the recipient, encoded in hex.  Can also use the
+/// key name `pubKey` for this.
+/// - `timestamp` (required) the timestamp of the message in unix epoch milliseconds, passed as an
+/// integer.  Timestamp may not be in the future (though a few seconds tolerance is permitted).  For
+/// backwards compatibility may be passed as a stringified integer.
+/// - `ttl` (required, unless expiry given) the message's lifetime, in milliseconds, passed as a string
+/// or stringified integer, relative to the timestamp.  Timestamp+ttl must not be in the past.  For
+/// backwards compatibility may be passed as a stringified integer.
+/// - `expiry` (required, unless ttl given) the message's expiry time as a unix epoch milliseconds
+/// timestamp.  (Unlike the above, this cannot be passed as an integer).
```
Comment (majestrate) [inline httpserver/client_rpc_endpoints.h]: as a stringified integer

**Pull request merged**